### PR TITLE
Added short version for most commonly used command line options

### LINKED ISSUE #11644: Command Line Switches: Addition of short switches for their long counterparts
### Is your feature request related to a problem? Please describe.

NVDA offers several command line switches that can be used. Some of these switches offer both short and long versions - could we possibly consider adding short versions of the following? This would have the potential to speed up the workflows of power users/system admins when manually typing out these switches at regular intervals:

--disable-addons - consider adding a "-noaddons" switch to perform this task

--debug-logging - consider adding a "-debuglog" switch to perform this task

--no-logging - consider adding a "-nolog" switch to perform this task

--no-sr-flag - consider adding a "-noflag" switch to perform this task

### LINKED ISSUE #17485: NVDA restart process is not robust
Issue found while trying to fix #11644.

NVDA restart process is not robust at all; there are many unforseen corner cases.

#### Steps to reproduce:
1. Start "`nvda --disable-add`" (please note the incomplete command line flag)
2. Press `NVDA+Q`, select "Restart" and validate
3. Repeat 2 as many times as wanted
#### Actual behavior:
After steps 2 and 3, NVDA restarts again with add-ons disabled, in contradiction to what is written in the restart dialog.
#### Expected behavior:
After steps 2 and 3, NVDA should restart with add-ons re-enabled.
#### Steps to reproduce:
1. Start "`nvda --lang=it --lang=en --disable-addons --disable-addons`"
2. Press `NVDA+Q`, select "Restart" and validate
3. Press `NVDA+Q`, select "Restart" and validate
#### Actual behavior:
After step 2, NVDA is restarted with add-ons disabled and English language.
#### Expected behavior:
After steps 2 and 3, NVDA should restart with add-ons re-enabled.
#### Steps to reproduce:
1. Start "`nvda --lan=it`"
2. Open NVDA settings dialogs
#### Actual behavior:
At step 1., no command parsing error window appears.
At step 2, NVDA appears in usual Windows language.

#### Expected behavior:
At step 2. NVDA should appear in Italian, since prefix matching should apply to command line options, even if not really advertised.
Alternatively, if such option writing is not allowed, at step 1., a command line error window should appear.

### NVDA logs, crash dumps and other attachments:
N/A
#### NVDA installed/portable/running from source:
Installed or from source
#### NVDA version:
2024.4.1
#### Windows version:
Windows 10 22H2 (AMD64) build **19045.5131**
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No, but probably no change
#### If NVDA add-ons are disabled, is your problem still occurring?
N/A
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tested, unrelated.

### PR DESCRIPTION
**Pull request by CyrilleB79**
### Link to issue number:
Fixes #11644
Fixes #17485
### Summary of the issue:
* Most common command line options need a short version.
* Command line options is not robust, leading to undesirable effects when restarting NVDA
### Description of user facing changes
* Short versions of two command line flags have been added: `-d` for `--disable-addons` and `-n` for `--lang`
* Using incomplete command line flags or duplicated command line options should not produce unexpected effects when restarting NVDA.

### Description of development approach
* Added new flags
* Check parsed app args rather than raw `sys.argv` to know the options that NVDA has actually taken into account.
* Removed `languageHandler.getLanguageCliArgs` since we do not use this function anymore and it was not correctly working as expected for some corner case (duplicated flag, incomplete flag)

### Testing strategy:
Manual tests from source:
* Start NVDA with `--lang`, `-n`, `--disable-addons`, `-d`.
* When language is forced from command line, check the language combo-box in general settings
* STR1, STR2 and STR3 from #17485
* Various other options to launch NVDA and then restarting it.

Also tested from launcher with options.

### Known issues with pull request:
None
### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [ ] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

@coderabbitai summary

### CODE REVIEW

**Comment by CyrilleB79** [comment]
@Adriani90 wrote:
> Does it make sense to also add -i, -is, -p and -ps for install, install silent, create portable copy and create portable copy silent? In corporate environments these are definitely commonly used.

Definitely not. These are not commonly used, except by IT guys in various install automation scripts. And in scripts the long version should be preferred because it is self explanatory. Short forms are useful only for regularly (and manually) typed command lines.
See Reef's and @seanbudd's comments in #11644 for the discussion.

Even a short version for `--debug-logging` has been refused by Sean, arguing that there is already the quite short `-l10`.

Also note that `-is` would be invalid, since short forms, i.e. flags beginning with only one dash, should use only one letter since they can be combined. E.g. "`nvda -sm`" is the same thing as "`nvda -s -m`".

**Comment by CyrilleB79** [comment]
@seanbudd, I am thinking of moving the parser code in a dedicated file, what would avoid the quite strange "`import __main__`". But for ease of review, I will do this once you have first reviewed the current code, and only if you agree with this move of course.

File: source/core.py [hunk 1]
```diff
@@ -214,6 +214,28 @@ class NewNVDAInstance:
 	directory: Optional[str] = None
 
 
+def computeRestartCLIArgs(removeArgsList: list[str] | None = None) -> list[str]:
+	"""Generate an equivalent list of CLI arguments from the values in globalVars.appArgs.
+	:param removeArgsList: A list of value to ignore when looking in globalVars.appArgs.
```
Comment (seanbudd) [inline source/core.py:219]:
```suggestion
	:param removeArgsList: A list of values to ignore when looking in globalVars.appArgs.
```

File: user_docs/en/changes.md [hunk 2]
```diff
@@ -46,6 +46,8 @@ To use this feature, "allow NVDA to control the volume of other applications" mu
 * Component updates:
   * Updated LibLouis Braille translator to [3.32.0](https://github.com/liblouis/liblouis/releases/tag/v3.32.0). (#17469, @LeonarddeR)
   * Updated CLDR to version 46.0. (#17484, @OzancanKaratas)
+* Short versions of the most commonly used command line Options have been added: `-d` for `--disable-addons` and `-n` for `--lang`.
+Prefix matching on command line flags, e.g. using `--di` for `--disable-addons` is no longer supported. (#11644, @CyrilleB79)
```
[the remark this answers is not in the document]
Comment (CyrilleB79) [reply user_docs/en/changes.md:50]: This change is not only visible by developers of NVDA or add-ons but by any user, e.g. sys admins, testers who do not need to read the changes for developers.

If you wish, I may add a copy of this information in the API-breaking changes section, what would be quite a duplicate. But I think think it's important to keep this information in the general "Changes" section.

[the remark this answers is not in the document]
Comment (seanbudd) [reply user_docs/en/changes.md:50]: yes at the very least a duplicate would be good

**Comment by seanbudd** [comment]
also feel free to move the code outside of nvda.pyw

**Comment by seanbudd** [comment]
In future @CyrilleB79 please avoid merging master in when moving code. It makes it much harder to review with a command like `git diff --color-moved afdceca92fb40a5c967eb10eca21b036e28173bb 96160b28e3b0ae61b9912999e9a55dfd00641e3b`

**Review by seanbudd — APPROVED** [review]
Thanks @CyrilleB79 - reviewed with `git diff -w --color-moved-ws=allow-indentation-change  --color-moved 05edaba0545baa44dfbf086b31d63dccb9df36d2 83f4b1489e057602c3f48c223dddcfaf8b5dfbba`

**Comment by CyrilleB79** [comment]
> In future @CyrilleB79 please avoid merging master in when moving code. It makes it much harder to review with a command like `git diff --color-moved afdceca92fb40a5c967eb10eca21b036e28173bb 96160b28e3b0ae61b9912999e9a55dfd00641e3b`

OK. Sorry for the complication.

**Comment by seanbudd** [comment]
@CyrilleB79 - just to clarify, was fine in this case, but if there was modifications/moves mixed with the merge it would be challenging
